### PR TITLE
Add pull request template for Pyodide Github repository [skip ci]

### PR DESCRIPTION
**Pull request by ryanking13**
This PR adds a pull request template.
(Some words are borrowed from [psf/black](https://github.com/psf/black/blob/main/.github/PULL_REQUEST_TEMPLATE.md))

The template contains:

- Description
- Checklists
  - adding changelog
  - adding tests
  - adding documentation
- **Notes about CI failure** (https://github.com/pyodide/pyodide/pull/1920#issuecomment-957870962)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     so don't be afraid of making a PR. To help make things go a bit more
+     smoothly we would appreciate that you go through this template. -->
+
+### Description
+
+<!-- Please explain what your PR is about:
+     - reasoning for the change
+     - some details of updated code
+     - any noteworthy choices to be aware of
+     If there is a relavant issues, please refer those by #<issue_id> -->
```
Comment (hoodmane) [inline .github/PULL_REQUEST_TEMPLATE.md:11]:
```suggestion
	Please refer to any related issues by #<issue_id> -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<!-- Thanks for contributing to Pyodide! Any improvements are welcome,
+     so don't be afraid of making a PR. To help make things go a bit more
+     smoothly we would appreciate that you go through this template. -->
```
Comment (hoodmane) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
<!-- Thank you for contributing to Pyodide! All improvements are welcome,
     so don't be afraid to make a PR. -->
```

**Review by hoodmane — APPROVED** [review]
Thanks @ryanking13! I made a bunch of picky language adjustments.

**Comment by ryanking13** [comment]
Applied suggestions, thanks for the coments! Language details are one of the most hardest parts to me 🤦

**Comment by rth** [comment]
Thanks @ryanking13!

**Pull request merged**